meson.build

```diff
@@ -55,21 +55,27 @@ relative_dir = run_command(
 # Strip relative path prefixes from the code if possible, otherwise hide them.
 # The following snippet is inspired by the SwayWM Project under the MIT license.
 add_project_arguments(
-    cc.first_supported_argument([
+    cc.first_supported_argument(
         '-fmacro-prefix-map=@0@='.format(relative_dir),
-        '-DHSE_REL_SRC_DIR="@0@"'.format(relative_dir),
-    ]),
+        '-DHSE_REL_SRC_DIR="@0@"'.format(relative_dir)
+    ),
     language: 'c'
 )
 
 if get_option('b_sanitize').contains('undefined')
     add_project_arguments(
-        cc.get_supported_arguments([
-            '-fno-sanitize-recover=undefined',
-            '-fsanitize-recover=alignment',
-        ]),
+        cc.get_supported_arguments('-fno-sanitize-recover=undefined'),
         language: 'c'
     )
+else
+    # userspace-rcu fails alignment in the following functions:
+    #   - _urcu_bp_read_lock
+    #   - _urcu_bp_read_unlock
+    #   - _urcu_bp_read_ongoing
+    #
+    # Ignoring them doesn't seem to work since defining _LGPL_SOURCE, forces
+    # them to be inline functions.
+    add_project_arguments('-D_LGPL_SOURCE', language: 'c')
```

Comment:
There was a reason in the past we didn't define this, check with Steve.  The alternative is that we had to link with urcu rather than build it in, which is slower but avoids licensing issues.  I can't recall the details....

Reply:
Since we no longer ship binaries maybe we can build urcu in?  Dunno...

```
Dynamic-only linking with the LGPL library is used if _LGPL_SOURCE is not
defined. It permits relinking with newer versions of the library, which is
required by the LGPL license.
```

Reply:
FYI this isn't new. I'm just moving it to be not defined if ubsan is enabled because of what I wrote in the comment.

I guess it is worth double checking though. @smoyergh is it ok if we define this? It should all be good since a user has the ability to build and relink HSE should they choose to do so.

Reply:
Isn't new as of when?  I don't recall ever defining that in our builds back when and prior to when we slogged through all this licensing stuff.  Things may be different now...

Reply:
At least when the new build system came along. I probably wouldn't have defined it if it wasn't in the CMake stuff though.

Reply:
For non-GPL code, it is considered a best practice to dynamically link LGPL libraries because there is some uncertainty around the implications of static linking and inlining.  FOSSA has a reasonably good discussion [here](https://fossa.com/blog/open-source-software-licenses-101-lgpl-license/).

So net is we're safer not defining _LGPL_SOURCE.

Reply:
> There is a kind of safe harbor for using LGPL code as a dynamically linked library; for statically linked libraries, a distributor must offer access to not only the library’s source code, but other information or materials necessary to rebuild the program.

Almost seems like we should have a build option for an `lgpl` friendly build that would allow the distributor to get the performance benefits if they can abide by the license.